SoftLayer/CLI/virt/dns.py

```diff
@@ -58,6 +72,38 @@ def sync_a_record():
             rec['ttl'] = ttl
             dns.edit_record(rec)
 
+    def sync_aaaa_record():
+        """Sync AAAA record."""
+        records = dns.get_records(zone_id,
+                                  host=instance['hostname'],
+                                  record_type='aaaa')
+
+        try:
+            # done this way to stay within 80 character lines
+            component = instance['primaryNetworkComponent']
+            record = component['primaryVersion6IpAddressRecord']
+            ip_address = record['ipAddress']
+        except KeyError:
+            raise exceptions.CLIAbort("%s does not have an ipv6 address"
+                                      % instance['fullyQualifiedDomainName'])
+
+        if not records:
+            # don't have a record, lets add one to the base zone
+            dns.create_record(zone['id'],
+                              instance['hostname'],
+                              'aaaa',
+                              ip_address,
+                              ttl=ttl)
+        else:
+            recs = [x for x in records if x['type'].lower() == 'aaaa']
```

Comment:
I was asking because you are already using a filter with the record_type argument and the A-record version doesn't. It appears like adding the filter made this line of code un-needed.